rasa/server.py

```diff
@@ -317,6 +321,8 @@ async def get_test_stories(
     else:
         trackers = [await processor.get_tracker(conversation_id)]
 
+    trackers = [get_events_before_restart(tracker) for tracker in trackers]
```

Comment:
Why is this required, does `Restarted` presence in `tracker.events` cause a bug?

Reply:
The presence of `Restarted` means the session is translated into a blank story  - due to using the applied events as defined [here](https://github.com/RasaHQ/rasa/blob/dc4930d50b800c6a059f87b3f92f27468d995aad/rasa/shared/core/trackers.py#L490-L491). Tbh, I think the more correct thing to do is to change the definition of how sessions are split to include restarted events at the beginning of each session instead of leaving them at the end - but I'm concerned this would result in changes elsewhere that we don't want.

Reply:
I see, so this becomes a problem further below in L340 because `tracker.as_story()` makes a call to `applied_events` which will be an empty list if the last event was `Restarted`?

Reply:
What would happen if other events follow the `Restarted` event?

Reply:
> so this becomes a problem further below in L340 because tracker.as_story() makes a call to applied_events which will be an empty list if the last event was Restarted?


Exactly.


> What would happen if other events follow the Restarted event?


That should not be possible in this context because `Restarted` is always followed directly by `SessionStarted`, and the conversation has already been split on `SessionStarted` events.